src/fandom-wirewax.plugin.js

```diff
@@ -0,0 +1,257 @@
+    var script = document.createElement("script");
+    script.src = fandomSDKUrl;
+    script.addEventListener("load", resolve);
+    script.addEventListener("error", function(e){reject(e.error)});
+    document.head.appendChild(script);
+  });
+};
+
+function FandomWirewaxPlugin(rootId, options) {
+  this.isPlayerRegistered = false;
+  this.rootId = rootId;
+  this.options = options;
+  this.player = options.player;
+  this.autoPlay = true;
+  this.setupEmbedder = this.setupEmbedder.bind(this);
+  this.registerEvents = this.registerEvents.bind(this);
+  this.startTimeUpdate = this.startTimeUpdate.bind(this); 
+  this.JWPlayHandler = this.JWPlayHandler.bind(this); 
+  this.JWPauseHandler = this.JWPauseHandler.bind(this); 
+  this.JWSeekHandler = this.JWSeekHandler.bind(this); 
+  this.WirewaxPlayHandler = this.WirewaxPlayHandler.bind(this);
+  this.WirewaxPauseHandler = this.WirewaxPauseHandler.bind(this); 
+  this.WirewaxSeekedHandler = this.WirewaxSeekedHandler.bind(this); 
+  this.WirewaxHotspotClickHandler = this.WirewaxHotspotClickHandler.bind(this);
+  this.WirewaxOverlayShowHandler = this.WirewaxOverlayShowHandler.bind(this);
+  this.WirewaxOverlayHideHandler = this.WirewaxOverlayHideHandler.bind(this);
+  this.WirewaxOverlayEventHandler = this.WirewaxOverlayEventHandler.bind(this);
+  this.startTimeUpdate = this.startTimeUpdate.bind(this);
+  this.stopTimeUpdate = this.stopTimeUpdate.bind(this);
+  this.hotspotEvent = false;
+  this.debouncePlayPause = false;
+
+  this.player.on("playlistItem", function(){
+    if (this.embedder) {
+
+      this.stopTimeUpdate();
+
+      // Dispose pre video interaction
+      try {
+        this.embedder.dispose();
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    // Search JW media id
+    var mediaId =
+      this.player.getConfig().playlistItem.mediaid ||
+      this.player.getConfig().playlistItem.videoId;
+
+    // validate interaction
+    if (!mediaId) {
+      throw new TypeError("No JW media id is specified.");
+    }
+
+    fetch(
+      'https://edge-player.wirewax.com/jwPlayerData/' + mediaId + '.txt'
+    )
+    .then(function(response){
+      if (response.status !== 200) {
+        throw new Error("No vidId is mapped with this mediaid");
+      }
+
+      return response.json();
+    })
+    .then(function(data){
+      this.vidId = data;
+
+      // Inject SDK
+      return injectEmbedderSDK();
+    }.bind(this))
+    .then(this.setupEmbedder)
+    .then(this.registerEvents)
+    .catch(function(error) {
+      console.warn(error);
+    });
+  }.bind(this));
+}
+
+FandomWirewaxPlugin.prototype.setupEmbedder = function () {
+  if (!this.embedder) {
+    // Create a container
+    this.container = document.createElement("div");
+    this.container.classList.add("vjs-wirewax-container");
+    this.container.setAttribute(
+      "style",
+      "position: absolute; height: 100%; width: 100%; top: 0; pointer-events: none"
```

Comment:
We are doing this inline like this because its the only item and easier than adding a .css file?

Reply:
Also the div being targeted only exists when wirewax is embedded.